### PR TITLE
[Enhancement] improve primary key table compaction schedule strategy

### PR DESCRIPTION
**Pull request by luohaha**
This PR contains:
1. When handle compaction apply task, it will consume memory in `update`, so we also need to consider it.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/olap_server.cpp|0|4|00.00%|[410, 411, 423, 424]|
|:large_blue_circle:|src/storage/tablet_updates.cpp|1|1|100.00%|[]|

**Review by meegoo — DISMISSED** [review]

File: be/src/common/config.h [hunk 1]
```diff
@@ -294,6 +294,8 @@ CONF_Int32(update_compaction_per_tablet_min_interval_seconds, "120"); // 2min
 CONF_mInt64(max_update_compaction_num_singleton_deltas, "1000");
 CONF_mInt64(update_compaction_size_threshold, "268435456");
 CONF_mInt64(update_compaction_result_bytes, "1073741824");
+// Timeout of waiting for apply after compaction commit, counters in ms.
+CONF_mInt64(update_compaction_wait_for_version_timeout_ms, "120000"); // 120s
```
Comment (chaoyli) [inline be/src/common/config.h:298]: If you and a timeout are configured, how to control the memory usage during 120s?

Comment (luohaha) [reply be/src/common/config.h:298]: We can increase `update_compaction_wait_for_version_timeout_ms`, let compaction wait apply finish, so compaction's apply task won't be pending in apply queue.

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/olap_server.cpp|0|2|00.00%|[410, 411]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/olap_server.cpp|0|2|00.00%|[410, 411]|

**Review by meegoo — APPROVED** [review]

**Pull request merged**